### PR TITLE
StoreKitConfigTestCase: replaced artificial delay with more and longe…

### PR DESCRIPTION
**Pull request by NachoSoto**
Follow up to #1017.

### CODE REVIEW

File: PurchasesTests/Purchasing/ProductsFetcherSK1Tests.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@ class ProductsFetcherSK1Tests: XCTestCase {
     var productsRequestFactory: MockProductsRequestFactory!
     var productsFetcherSK1: ProductsFetcherSK1!
 
-    private static let defaultTimeout: DispatchTimeInterval = .seconds(30)
```
Comment (NachoSoto) [inline PurchasesTests/Purchasing/ProductsFetcherSK1Tests.swift:11]: These are unit tests so they don't need such a long delay.

File: StoreKitUnitTests/StoreKitConfigTestCase.swift [hunk 2]
```diff
@@ -36,7 +38,7 @@ class StoreKitConfigTestCase: XCTestCase {
         testSession.disableDialogs = true
         testSession.clearTransactions()
 
-        self.waitForStoreKitTestIfNeeded()
+//        self.waitForStoreKitTestIfNeeded()
```
Comment (NachoSoto) [inline StoreKitUnitTests/StoreKitConfigTestCase.swift:41]: If this approach works I'll remove this whole thing
(And let's not forget to remove the environment variable).

**Comment by NachoSoto** [comment]
Tests passed. Running again to verify.

**Comment by NachoSoto** [comment]
Here's what [an Apple engineer said](https://developer.apple.com/forums/thread/682404) about
```
The connection to service named com.apple.storekitservice.sim2host was invalidated: failed at lookup with error 3 - No such process.
```

> When targeting the sandbox environment, the Product.products(for:) method is only supported in simulators when you are running Xcode 13 on macOS 12 Beta 4 or newer.

I can't find it, but I'm guessing CircleCI isn't running Monterey?

**Comment by aboedo** [comment]
@NachoSoto I believe they're referring to `Product.products(for:)`, which is SK2-only. If you're just doing plain old `SKProductsRequests`, though, that's SK1

**Comment by aboedo** [comment]
oh nvmd I just read the name of the branch, but yes, those are failing for SK2

**Comment by NachoSoto** [comment]
Yeah, which we use in `PurchasesOrchestratorTests.fetchSk2Product`, which failed as well. And since we're using that directly there's no retry mechanism there.

**Comment by aboedo** [comment]
the weird part though is that I'd expect that if the issue is OS compatibility, it'd fail 100% of the time

**Comment by NachoSoto** [comment]
Unless what the Apple engineer was talking about is the fact that the problem with the _daemon_ is solved in Monterey.

**Comment by NachoSoto** [comment]
This is pretty frustrating. We have flaky tests because of `StoreKit` even with the 30 second delay: https://app.circleci.com/pipelines/github/RevenueCat/purchases-ios/4337/workflows/88512340-1833-4bae-a132-376d6f8cf0e6/jobs/12872 (that's in an unrelated branch with none of these changes).

File: Purchases/FoundationExtensions/DispatchTimeInterval+Extensions.swift [hunk 3]
```diff
@@ -27,6 +27,21 @@ extension DispatchTimeInterval {
...
+        }
+    }
+
 }
 
 // swiftlint:enable identifier_name
+
+func + (lhs: DispatchTimeInterval, rhs: DispatchTimeInterval) -> DispatchTimeInterval {
+    return .milliseconds(Int(lhs.milliseconds + rhs.milliseconds))
+}
```
Comment (aboedo) [inline Purchases/FoundationExtensions/DispatchTimeInterval+Extensions.swift:47]: ❤️

File: Purchases/Purchasing/StoreKit1/ProductsFetcherSK1.swift [hunk 4]
```diff
@@ -26,7 +26,7 @@ class ProductsFetcherSK1: NSObject {
     private var completionHandlers: [Set<String>: [Callback]] = [:]
     private let requestTimeout: DispatchTimeInterval
 
-    private static let numberOfRetries: Int = 5
+    private static let numberOfRetries: Int = 10
```
Comment (aboedo) [inline Purchases/Purchasing/StoreKit1/ProductsFetcherSK1.swift:29]: Perhaps we should document somewhere in the class that this is going to return after whichever comes first of: 
- a success
- retries up to `numberOfRetries`
- timeout of 30 secs

I think the code is clear, but just to save someone from actually looking through it if they wonder how the class works

**Review by aboedo — APPROVED** [review]
fingers crossed

File: Purchases/Purchasing/StoreKit1/ProductsFetcherSK1.swift (hunk 4 above)
Comment (NachoSoto) [reply Purchases/Purchasing/StoreKit1/ProductsFetcherSK1.swift:29]: Done.

**Pull request merged**